tests/utils/test_io.py

```diff
@@ -49,6 +50,24 @@ def test_load_model_checkpoint(path):
     model = load_model(path)
     assert type(model) is SoapBpnn
 
+
+@pytest.mark.parametrize(
+    "path",
+    [
+        RESOURCES_PATH / "model-64-bit.ckpt",
+    ],
+)
+def test_load_model_checkpoint_wrong_version(path):
+    file = RESOURCES_PATH / "model-64-bit-version5000000.ckpt"
+    model = torch.load(path, weights_only=False, map_location="cpu")
+    model["architecture_version"] = 5000000
+    torch.save(model, file)
+    with pytest.raises(
+        NotImplementedError,
+        match="Checkpoint upgrade is not implemented for the SOAP-BPNN model.",
+    ):
+        load_model(file)
+
     # TODO: test that weights are the expected if loading with `context == 'export'`.
```

Comment:
this should be left in the function above, it is a comment for `test_load_model_checkpoint`